src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -125,7 +125,7 @@ private set
         private sealed class SyntheticBinderImpl : BuckStopsHereBinder
         {
             private readonly SyntheticBoundNodeFactory _factory;
-            internal SyntheticBinderImpl(SyntheticBoundNodeFactory factory) : base(factory.Compilation)
+            internal SyntheticBinderImpl(SyntheticBoundNodeFactory factory) : base(factory.Compilation, associatedSyntaxTree: null)
```

Comment:
>associatedSyntaxTree: null [](http://example.com/codeflow?start=104&length=26)

What's the impact of not having an associated syntax tree in this binder?

Reply:
This binder is only used to bind synthesized invocation expressions of compiler-known helper methods, it's only employed once the container of the method is bound. We don't expect compiler-known helpers to be file scoped so it seems OK to give a null syntax tree here.